src/components/ContentOpenWith/ContentOpenWith.js

```diff
@@ -187,16 +200,90 @@ class ContentOpenWith extends PureComponent<Props, State> {
             .getOpenWithIntegrations(fileId, language, this.fetchOpenWithSuccessHandler, this.fetchErrorHandler);
     }
 
+    setResolvedIntegrationState = (integrations: Array<Integration>) =>
+        this.setState({ integrations, isLoading: false });
+
     /**
      * Fetch app integrations info needed to render.
      *
      * @param {OpenWithIntegrations} integrations - The available Open With integrations
      * @return {void}
      */
-    fetchOpenWithSuccessHandler = (integrations: Array<Integration>): void => {
+    fetchOpenWithSuccessHandler = async (integrations: Array<Integration>): Promise<any> => {
+        const boxEditIntegration = integrations.find(({ appIntegrationId }) =>
+            this.isBoxEditIntegration(appIntegrationId),
+        );
+
+        if (!boxEditIntegration || boxEditIntegration.isDisabled) {
+            this.setState({ integrations, isLoading: false });
+            return;
+        }
+
+        try {
+            const { extension } = await this.getIntegrationFileExtension();
+            boxEditIntegration.extension = extension;
+        } catch (error) {
+            boxEditIntegration.disabledReasons.push(
+                <FormattedMessage {...messages.executeIntegrationOpenWithErrorHeader} />,
+            );
+            boxEditIntegration.isDisabled = true;
+            this.setState({ integrations, isLoading: false });
+            return;
+        }
+
+        // If Box Edit is present and enabled, we need to set its ability to locally open the given file
+        if (!(await this.checkBoxEditAvailability())) {
+            boxEditIntegration.disabledReasons.push(<FormattedMessage {...messages.boxToolsUninstalledErrorMessage} />);
+            boxEditIntegration.isDisabled = true;
+        } else if (!(await this.canOpenExtensionWithBoxEdit(boxEditIntegration))) {
+            boxEditIntegration.disabledReasons.push(<FormattedMessage {...messages.boxToolsBlacklistedError} />);
+            boxEditIntegration.isDisabled = true;
+        }
+
         this.setState({ integrations, isLoading: false });
     };
 
+    /**
+     * Fetches the file extension of the current file.
+     *
+     * @return {Promise}
+     */
+    getIntegrationFileExtension = (): Promise<BoxItem> => {
+        const { fileId }: Props = this.props;
+        return new Promise((resolve, reject) => {
+            this.api.getFileAPI().getFileExtension(fileId, resolve, reject);
+        });
+    };
+
+    /**
+     * Uses Box Edit to check if Box Tools is installed and reachable
+     *
+     * @param {Integration} boxEditIntegration - A Box Edit or Box Edit SFC integration
+     * @return {void}
+     */
+    checkBoxEditAvailability = (): Promise<boolean> => {
```

Comment:
The other similar method begins with `can` below. Since the promise here always resolves with a boolean, might be good to stay consistent.

Reply:
switching to a similar format. `isBoxEditAvailable`